### ISSUE TITLE
False Error

### ISSUE THREAD

**Issue by jasonwadsworth** [aws-cloudformation/cfn-python-lint#1485]
*cfn-lint version: (`cfn-lint --version`)*
0.29.5

*Description of issue.*
I'm getting the following error in a [`RequestMappingTemplate`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-resolver.html#cfn-appsync-resolver-requestmappingtemplate) on [`AWS::AppSync::Resolver`](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-appsync-resolver.html). I believe this should be an exception to the rule, as it's completely normal to have `${}` inside the mapping template.

>Found an embedded parameter "${context.arguments.clientId}" outside of an "Fn::Sub" at Resources/ListMemberAttributeDefinitionsResolver/Properties/RequestMappingTemplate

Example Resource:
```
  Resolver:
    Type: AWS::AppSync::Resolver
    Properties:
      ApiId: !ImportValue AppSyncApiId
      TypeName: Query
      Kind: UNIT
      FieldName: myFieldName
      DataSourceName: !GetAtt AppSyncDataSource.Name
      RequestMappingTemplate: |-
        {
            "version": "2018-05-29",
            "method": "GET",
            "params": {
                "headers": $utils.toJson($util.map.copyAndRemoveAllKeys($context.request.headers, ['host','connection','user-agent','expectation','transfer-encoding','content-length'])),
            },
            "resourcePath": "/some-path/${context.arguments.someId}"
        }
      ResponseMappingTemplate: |-
        ## return the body
        #if($context.result.statusCode == 200)
             $ctx.result.body
        #else
            ##if response is not 200, append the response to error block.
            $util.error($context.result.body, "$context.result.statusCode")
        #end
```
